python/tank/deploy/tank_commands/get_entity_commands.py

```diff
@@ -0,0 +1,274 @@
+                                  key. Each value is a list of commands. A
+                                  command is a dictionary with the following
+                                  format:
+                                    {
+                                      "name":        command to execute
+                                      "title":       title to display for the
+                                                     command
+                                      "description": description of what the
+                                                     command does
+                                      "icon":        path to the icon of this
+                                                     command
+                                    }""",
+                "type":        "dict"
+            }
+        }
+
+    def run_interactive(self, log, args):
+        """
+        Tank command accessor
+
+        :param log: std python logger
+        :param args: command line args
+        """
+        raise TankError("This Action does not support command line access")
+
+    def run_noninteractive(self, log, parameters):
+        """
+        Tank command API accessor.
+        Called when someone runs a tank command through the core API.
+
+        :param log: std python logger
+        :param parameters: dictionary with tank command parameters
+        """
+        pipeline_config_path = parameters["configuration_path"]
+        entities = parameters["entities"]
+
+        # at the moment, the caching mechanism works with the entity types,
+        # so we group by type and fetch the commands for each type
+        per_entity_type = itertools.groupby(entities, operator.itemgetter(0))
+
+        commands_per_entity = {}
+        for (entity_type, entities_of_type) in per_entity_type:
+            # make a list out of the grouped entity tuples
+            entities_of_type = list(entities_of_type)
+            try:
+                cache_content = self._load_cached_data(pipeline_config_path,
+                                                       entity_type)
+                commands = self._parse_cached_commands(cache_content)
+
+                # at the moment, the commands are the same for all entities of
+                # that type
+                for entity in entities_of_type:
+                    commands_per_entity[entity] = commands
+            except TankError, e:
+                log.error("Failed to fetch the commands from the Pipeline "
+                          "Configuration at '%s' for the entity type %s.\n"
+                          "Details: %s"
+                          % (pipeline_config_path, entity_type, e))
+
+        return commands_per_entity
+
+    def _get_cache_name(self, platform, entity_type):
+        """
+        Constructs the expected name for the cache file of a particular entity
+        type.
+
+        :param platform:    platform that will use the cached information
```

Comment:
this is not clear. better to say that it is `sys.platform` style strings or something like that.